### PR TITLE
Add ClickHouse module

### PR DESCRIPTION
**Pull request by VladRassokhin**
Rebase of #498 onto master with additional changes

### CODE REVIEW

File: modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package org.testcontainers.containers;
+
+import org.testcontainers.containers.wait.HttpWaitStrategy;
+
+import java.time.Duration;
+
+public class ClickHouseContainer<SELF extends ClickHouseContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
```
Comment (bsideup) [inline modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: there is no need for self-typing here

Comment (VladRassokhin) [reply modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: I don't get it. Almost all other JDBC containers have 'SELF', why this one shouldn't?

Comment (bsideup) [reply modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: Consider it a legacy :)

**Comment by bsideup** [comment]
@VladRassokhin @tolkonepiu merged, thanks!

**Pull request merged**